bokehjs/src/lib/models/annotations/info_pane.ts

```diff
@@ -0,0 +1,207 @@
+    }
+
+    const {anchor} = this.model
+    this.el.classList.remove(bk_right)
+    this.el.classList.remove(bk_left)
+    this.el.classList.remove(bk_above)
+    this.el.classList.remove(bk_below)
+
+    const arrow_size = 10  // XXX: keep in sync with less
+
+    display(this.el)
+
+   // slightly confusing: side "left" (for example) is relative to point that
+   // is being annotated but CS class ".bk-left" is relative to the tooltip itself
+
+    let top: number = 0
+    let left = 0
+    let right = 0
+
+    switch (anchor) {
+      case "left":
+        this.el.classList.add(bk_right)
+        left = x + (this.el.offsetWidth - this.el.clientWidth)
+        top = y - this.el.offsetHeight/2
+        break
+      case "right":
+        this.el.classList.add(bk_left)
+        right = (this.plot_view.layout.bbox.width - x)
+        top = y - this.el.offsetHeight/2
+        break
+      case "above":
+        this.el.classList.add(bk_below)
+        top = y + (this.el.offsetHeight - this.el.clientHeight) - arrow_size
+        left = Math.round(x - this.el.offsetWidth/2)
+        break
+      case "below":
+        this.el.classList.add(bk_above)
+        top = y - this.el.offsetHeight + arrow_size
+        left = Math.round(x - this.el.offsetWidth/2)
+        break
+    }
+
+    if (this.model.show_arrow)
+      this.el.classList.add(bk_tooltip_arrow)
+
+    if (this.el.childNodes.length > 0) {
+      this.el.style.top = `${top}px`
+      this.el.style.left = left ? `${left}px` : 'auto'
+      this.el.style.right = right ? `${right}px` : 'auto'
+    } else
+      undisplay(this.el)
+  }
+}
+
+export namespace InfoPane {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Annotation.Props & {
+    anchor: p.Property<TooltipAttachment>
+    //can be ignored if inner_only property isnt useful
+    inner_only: p.Property<boolean>
+    show_arrow: p.Property<boolean>
+    x: p.Property<number>
+    y: p.Property<number>
+    anchor_units: p.Property<SpatialUnits>
+    //data has to be able to get list of strings
+    data: p.Property<[string][]>
+    custom: p.Property<boolean>
+  }
+}
+
+export interface InfoPane extends InfoPane.Attrs {}
+
+export class InfoPane extends Annotation {
+  properties: InfoPane.Props
+  __view_type__: InfoPaneView
+
+  constructor(attrs?: Partial<InfoPane.Attrs>) {
+    super(attrs)
+  }
+
+  static init_InfoPane(): void {
+    this.prototype.default_view = InfoPaneView
+
+    this.define<InfoPane.Props>({
+      //is kept as is if the horizontal\vertical functionality shall come in use
+      anchor:       [ p.TooltipAttachment, 'horizontal'],
+      inner_only:   [ p.Boolean,           true        ],
+      show_arrow:   [ p.Boolean,           true        ],
+      x:            [ p.Number,             0          ],
+      y:            [ p.Number,             0          ],
+      anchor_units: [ p.SpatialUnits,      'data'      ],
+      data:         [ p.Array,                         ],
```

Comment:
In @mattpap other PRs he renamed `data` to `content` which I had also considered suggesting on this PR. I agree with @mattpap that changing to `content` is a better property name, and we should take the decision to change it now, since we have that freedom to pick whatever we want for a new model.

Reply:
👍 for `content`

Reply:
Yeah, that sounds good. I will update it.